fetch/programmer-humour/javascript.js

```diff
@@ -0,0 +1,39 @@
+// Function to fetch the latest comic
+function fetchLatestComic() {
+  // Set the loading message
+  document.getElementById('loading').style.display = 'block';
+  document.getElementById('comic-container').innerHTML = ''; // Clear previous comic if any
+  document.getElementById('error-message').innerHTML = ''; // Clear previous error message
+
+  fetch('https://xkcd.now.sh/?comic=latest')
+    .then(response => {
+      // Check if the response is successful (status 200)
+      if (!response.ok) {
+        throw new Error('Network response was not ok');
+      }
+      return response.json();
+    })
+    .then(data => {
+      // Log the received data to the console
+      console.log(data);
+
+      // Get the image URL from the response data
+      const imageUrl = data.img;
+
+      // Render the image in the DOM
+      const comicImage = `<img src="${imageUrl}" alt="Latest XKCD Comic">`;
```

Comment:
Since the value of `imageUrl` is coming from another website, there is a danger that its value can corrupt the generated string. (For details, you can ask ChatGPT `In JS, why is constructing HTML tag using template string is dangerous?`)

It is a safer practice to create `<img>` element (or any other HTML element) using the `document.createElement()` approach.